src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/AppointmentEventPublisher.kt

```diff
@@ -4,18 +4,27 @@ import org.springframework.context.ApplicationEvent
 import org.springframework.context.ApplicationEventPublisher
 import org.springframework.stereotype.Component
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.LocationMapper
-import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller.ActionPlanSessionController
-import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ActionPlanSession
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller.ReferralController
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Appointment
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AppointmentType
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.NotifyAppointmentService
 
 enum class AppointmentEventType {
   ATTENDANCE_RECORDED,
   BEHAVIOUR_RECORDED,
   SESSION_FEEDBACK_RECORDED,
 }
 
-class AppointmentEvent(source: Any, val type: AppointmentEventType, val actionPlanSession: ActionPlanSession, val detailUrl: String, val notifyPP: Boolean) : ApplicationEvent(source) {
+class AppointmentEvent(
+  source: Any,
+  val type: AppointmentEventType,
+  val appointment: Appointment,
+  val detailUrl: String,
+  val notifyPP: Boolean,
+  val appointmentType: AppointmentType
```

Comment:
This class is only used for SupplierAssessmentAppointments. I don't think AppointmentType is necessary here.
I suggest we rename this class to SupplierAssessmentAppointmentEventPublisher.

The equivalent for action plan appointments are `ActionPlanAppointmentEventPublisher`.